libraries/redisio.rb

```diff
@@ -50,4 +50,45 @@ def self.version_to_hash(version_string)
       rc: version_array[3]
     }
   end
+
+  # Load password from vault or data bag
+  #
+  # @param [Hash] opts defines where to get the secret
+  # @option opts [String] :chef_vault_name chef vault name to load secret from
+  # @option opts [String] :chef_vault_item chef vault item to load secret from
+  # @option opts [String] :chef_vault_key chef vault key to load secret from
+  # @option opts [String] :data_bag_name data bag name to load secret from
+  # @option opts [String] :data_bag_item data bag item to load secret from
+  # @option opts [String] :data_bag_key data bag key to load secret from
+  # @option opts [String] :data_bag_secret path to secret to decrypt data bag item
+  def self.load_secret(opts = {})
```

Comment:
Is there a reason we need to pass so many options? I feel like if we follow the path outlined here it would be much simpler.

Reply:
The way I added this follows the existing implementation in the providers

```
-      if current['data_bag_name'] && current['data_bag_item'] && current['data_bag_key']
```

I think the only way I can simplify this is the opts hash accepts `bag_name`, `item_name`, `item_key`, and `secret` and has defaults set for each. Then try to decrypt as a vault item or encrypted data bag item depending on how the specified item looks onthe chef server (i think chef-vault provides something like `is_vault_item?(bag, item)`.

Im just not sure if that gains much rather than how I have it now in being explicit in how the user tells the provider to find the password

Reply:
Why do we need to pass the secret in the first place? We should just rely on the user configuring their client to have the secret via `encrypted_data_bag_secret` in `/etc/chef/client.rb`.

https://docs.chef.io/config_rb_client.html

If you want to see a cleaner way I just merged this a few days ago: https://github.com/sensu/uchiwa-chef/pull/64

Reply:
Many people use more than one secret file to give a node access to multiple encrypted data bag items. We dont encrypt all of our encrypted data bag items with the same secret, so this needs to be specifiable.

Reply:
fair enough, honestly I generally rely on more extensive secret management than encrypted data bags so I wrongly assumed anyone who had strict security requirements would do something similar and not go through all that trouble to essentially create RBAC through multiple secrets.